x/expiration/types/expiration.go

```diff
@@ -0,0 +1,109 @@
+package types
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func NewExpiration(
+	moduleAssetID string,
+	owner string,
+	time time.Time,
+	deposit sdk.Coin,
+	message types.Any,
+) *Expiration {
+	return &Expiration{
+		ModuleAssetId: moduleAssetID,
+		Owner:         owner,
+		Time:          time,
+		Deposit:       deposit,
+		Message:       message,
+	}
+}
+
+// ValidateBasic basic format checking of the data
+func (e *Expiration) ValidateBasic() error {
+	now := time.Now()
```

Comment:
Nope. Nope. NoNoNoNoNoNo.

You cannot use `time.Now()` anywhere, ever.

This `ValidateBasic` function simply cannot do any checking on the expiration time other than maybe making sure it's not nil or 0. That validation will need to be done elsewhere where the block time can be accessed.

Reply:
Unresolving this until I see it fixed in the diff since it's such a necessity to have fixed.